icu4c/source/i18n/collationbuilder.cpp

```diff
@@ -307,11 +307,14 @@ CollationBuilder::addReset(int32_t strength, const UnicodeString &str,
             parserErrorReason = "normalizing the reset position";
             return;
         }
-        cesLength = dataBuilder->getCEs(nfdString, ces, 0);
+        cesLength = dataBuilder->getCEs(nfdString, ces, 0, errorCode);
         if(cesLength > Collation::MAX_EXPANSION_LENGTH) {
             errorCode = U_ILLEGAL_ARGUMENT_ERROR;
             parserErrorReason = "reset position maps to too many collation elements (more than 31)";
             return;
+        } else if (errorCode == U_INPUT_TOO_LONG_ERROR) {
```

Comment:
Just a question/comment: Can getCEs return any other errors? (And stylistically I like putting the error handling first, though I assume it does not make any practical difference here or the other similar places since getsCEs returns 0 if error).

Reply:
>Just a question/comment: Can getCEs return any other errors?

Come to think of it, it can't.  I checked the specific error code because I worried there might be more errors in the future, and then the error string this function is fabricating would be wrong.  But I'll change it.

>(And stylistically I like putting the error handling first, though I assume it does not make any practical difference here or the other similar places since getsCEs returns 0 if error).

I see your point.  I'll change this too.